content/n00bzctf_2023/johnDoesStrikesAgain.md

```diff
@@ -0,0 +1,138 @@
+---
+title: John Does Strikes Again
+date: 2023-06-10
+tags:
+  - osint
+---
+
+___
+
+## Description
+
+John Doe has escaped our high secruity[^1] prison again! We managed to intercept an xor key that he uses to send
+encrypted
+messages to people! Your aim is to find classified information on his top secret website! Start with the encrypted
+message -`b'\x13\x00\x1d-A*!\x00Q\x16R\x02\x12\x07\n\x1b>\x0e\x06\x1a~O-D CU\t\x0e\x06 E2\n\x17bA#\x0b\t>O\x11\x011O\tH*\x1b\x10-\x08\x00)E\x02\nMck~)\x07"\x01H*+\n_\x01\x00\x00\x00c\n\x00!\x12V\r\x1d4A\x19\x16\x0b"O!N(\x00\x13Dy\x02\x000\x08\rn\x16\x19E\x16,\x0fS\x17H+\x1c\x03N)\nEU1\x0e\x01c\x10\x1b+\x16\x02\x0c\x1d-A\x11\x15\r8\x16H\x0f#\x0e\x0cOx'`
+and the secret key -`YouCanNeverCatchJohnDoe!`We also intercepted the name of his
+account -`31zdugxvkayexc4hzqhixxcfxb4y`
+
+Author: NoobMaster
+
+## Solution
+
+Let's start by decrypting the message provided with the key `YouCanNeverCatchJohnDoe!`. The message is obvisouly in
+bytes, so we need to
+convert it to string and apply the XOR decryption by using the key as suggested in the challenge description.
+
+```python
+
+import itertools
+
+key = "YouCanNeverCatchJohnDoe!"
+message = b'\x13\x00\x1d-A*!\x00Q\x16R\x02\x12\x07\n\x1b>\x0e\x06\x1a~O-D CU\t\x0e\x06 E2\n\x17bA#\x0b\t>O\x11\x011O\tH*\x1b\x10-\x08\x00)E\x02\nMck~)\x07"\x01H*+\n_\x01\x00\x00\x00c\n\x00!\x12V\r\x1d4A\x19\x16\x0b"O!N(\x00\x13Dy\x02\x000\x08\rn\x16\x19E\x16,\x0fS\x17H+\x1c\x03N)\nEU1\x0e\x01c\x10\x1b+\x16\x02\x0c\x1d-A\x11\x15\r8\x16H\x0f#\x0e\x0cOx'
+
+decrypted_message = bytearray([m ^ k for (m, k) in zip(message, itertools.cycle(key.encode()))])
+
+print(decrypted_message.decode())
+
+```
+
+The output is the following: `John Doe: You know how much I love music so don't ask me that question every again! `
```

Comment:
the only thing, is that this inline code has a trailing space at the end lol

Reply:
Thx fixed :)

Reply:
@Stefan-Radu you have the approved the changement I made so I can merge